src/lsm/manifest_level.zig

```diff
@@ -38,6 +38,22 @@ pub fn ManifestLevelType(
 
         pub const Tables = SegmentedArray(TableInfo, NodePool, table_count_max, .{});
 
+        pub const CompactionTableRange = struct {
+            table: *TableInfo,
+            range: CompactionRange,
+        };
+
+        pub const CompactionRange = struct {
+            /// The total number of tables in the compaction across both levels, always at least 1.
+            table_count: usize,
+            /// The minimum key across both levels.
+            key_min: Key,
+            /// The maximum key across both levels.
+            key_max: Key,
+            // References to tables in level B that intersect with the chosen table in level A.
+            tables: std.BoundedArray(*TableInfo, constants.lsm_growth_factor),
+        };
```

Comment:
Mhmmm, I had observed this as well -- ManifestLevel is currently oblivious of all compaction logic. I was wondering if that was by design. :) 

ManifestLog is oblivious of compaction as well, except for [this](https://github.com/tigerbeetledb/tigerbeetle/blob/1b64cac51bc103db6c673ad59dc69d7c12f04e6e/src/lsm/manifest_log.zig#L85-L97) small logic.

I am on board with keeping ManifestLevel oblivious of compaction and moving these out to `compaction_range.zig`. Another option could be moving them to `compaction.zig`?

Reply:
However, on second thought -- _compaction_range_ finds all tables in Level X that intersect with a given range. This fits the bill of a generic ManifestLevel operation rather than something strictly related to compaction... 

_compaction_table_, on the other hand, is definitely logic very specific to compaction as it implements our table selection strategy.

Reply:
Great to think this through from different angles like this:
* To see where we can keep inner abstractions oblivious of outer abstractions.
* To move in-memory only logic from abstractions that also do I/O, to abstractions that only do in-memory logic for easier testing.

Perhaps the way forward is to keep these in Manifest, but to improve the names to be more descriptive, since they could also be seen as generic manifest utilities that compaction uses:
* `table_with_least_overlap_in_the_next_level()`
* `table_range()` or `overlapping_tables()` (or a name that also works well as a noun so that it can be used for the type result)

Reply:
@jorangreef @sentientwaffle as part of [this](https://github.com/tigerbeetledb/tigerbeetle/pull/875/commits/1132d35312a948249330de01ebf94dd2044a123c) commit, I've made the functions in ManifestLevel more generic, they're now [ManifestLevel.table_with_least_overlap](https://github.com/tigerbeetledb/tigerbeetle/blob/1132d35312a948249330de01ebf94dd2044a123c/src/lsm/manifest_level.zig#L464) and [ManifestLevel.overlapping_tables](https://github.com/tigerbeetledb/tigerbeetle/blob/1132d35312a948249330de01ebf94dd2044a123c/src/lsm/manifest_level.zig#L584). 

[Tree.compact_start_tables](https://github.com/tigerbeetledb/tigerbeetle/blob/1132d35312a948249330de01ebf94dd2044a123c/src/lsm/tree.zig#L721) still invokes [Manifest.compaction_table](https://github.com/tigerbeetledb/tigerbeetle/blob/1132d35312a948249330de01ebf94dd2044a123c/src/lsm/manifest.zig#L424) → [ManifestLevel.table_with_least_overlap](https://github.com/tigerbeetledb/tigerbeetle/blob/1132d35312a948249330de01ebf94dd2044a123c/src/lsm/manifest_level.zig#L464) → [ManifestLevel.overlapping_tables](https://github.com/tigerbeetledb/tigerbeetle/blob/1132d35312a948249330de01ebf94dd2044a123c/src/lsm/manifest_level.zig#L584).

Reply:
> Perhaps the way forward is to keep these in Manifest, but to improve the names to be more descriptive, since they could also be seen as generic manifest utilities that compaction uses:

@jorangreef Can you help me understand why these belong in the `Manifest`? Neither depends at all on the disk/IO.
Or did you mean `ManifestLevel`?

Reply:
I assumed Joran meant ManifestLevel even though he typed Manifest; but that's just because we had a discussion about moving the purely in-memory functionality from Manifest → ManifestLevel in our one-on-one yesterday morning. I hope I'm right! 😅

Reply:
Yeah `ManifestLevel` makes sense to me, especially since the naming is now generalized away from compaction.